vllm/v1/engine/async_llm.py

```diff
@@ -74,9 +78,26 @@ def __init__(
             usage_context=usage_context,
             multiprocess_mode=True,
             asyncio_mode=True,
+            log_stats=self.log_stats,
         )
 
-        self.output_handler = None
+        # Async tasks that run in the background.
+        # We don't initialize the handlers until we have a request so
+        # they are None first.
+        self.output_handler: Optional[asyncio.Task] = None
+        self.stats_handler: Optional[asyncio.Task] = None
+
+        if vllm_config.observability_config.log_stats:
+            # The stats manager will be invoked from different threads:
+            # 1. stats_handler: a background busy event loop that pulls stats
+            # from the EngineCore periodically. The stats logging also happens
+            # on the background thread.
+            # 2. the main event loop when a request is added (and input
+            # processed).
+            self.stats_manager = ThreadSafeEngineStatsManager(
```

Comment:
There are not any threads in this class (just `asyncio`) --- so we don't need a threadsafe variant.